src/app/components/database-seed/database-seed.component.ts

```diff
@@ -1,66 +1,214 @@
 import { Component, OnInit } from '@angular/core';
-import { forkJoin } from 'rxjs';
-import { mergeMap, tap } from 'rxjs/operators';
+import { forkJoin, Observable, of } from 'rxjs';
+import { map, mapTo, mergeMap, switchMap, tap } from 'rxjs/operators';
+import { merge as _merge, omit as _omit } from 'lodash';
```

Comment:
Do we plan on updating Lodash as newer versions come out? According to the creator, [`omit` will be deprecated in v5 due to poor performance](https://github.com/lodash/lodash/issues/2930#issuecomment-272298477).  We may need to rewrite the code if so.

Reply:
We will update libraries every month to please the Dependabot God, unless the update of a given library is not trivial. I learned about the existence of `@babel/plugin-proposal-object-rest-spread` by following the link you shared. I don't find this approach suitable here as it depends on the order of the object properties. I decided to go with `_.pick` and specify the properties to keep (array). I wasn't sure about how `_.pick` handled array that include property names that are not in a given object, for example `Challenge.endDate` is optional. `_.pick` behave nicely and acts has a filter and it will not complain if a given object instance does not have the property. I'm still - and always - open to alternatives that you would like to share.

Reply:
I like the `_.pick` approach as well! That being said, it may become quite tedious if numerous properties are needed... do you think that could happen?  If so, we could always write a util function that mimics `_.omit`?  I came across this article [here](https://levelup.gitconnected.com/omit-is-being-removed-in-lodash-5-c1db1de61eaf) that showcases different alternatives and their performances.

Reply:
Especially since we still are in the early stage of development, it's likely that object properties will be updated. It's also more likely that we will add or rename properties - i.e. which will need us to update the seeding code - than removing properties from `{schema}` object to get `{schema}CreateResponse`. Also one thing that I don't like about `_.pick` is its behavior as I described earlier can lead to a point where the list of properties specified is out of sync with the properties of `{schema}CreateResponse`.

> We can throw away the pickBy option as it it too slow and doesn’t provide any benefits.

Assuming that `pick` has similar performance, then let's use one of the alternative proposed. I just updated the code to an an `omit` function based on ESRest + JavaScript delete. This implementation does not allow to remove flatten paths, which should be fine for now.